test/helpers/test_provisioner.rb

```diff
@@ -65,6 +65,11 @@ def deploy_metric_server
         logger: logger, log_failure_by_default: true, default_timeout: '5s')
 
       Dir.glob("test/setup/metrics-server/*.{yml,yaml}*").map do |resource|
+        if kubectl.server_version < Gem::Version.new('1.11.0') && resource =~ /metrics-server-deployment.yaml/
+          next
+        elsif kubectl.server_version >= Gem::Version.new('1.11.0') && resource =~ /metrics-server-deployment_021.yaml/
+          next
+        end
         found = kubectl.run("get", "-f", resource, log_failure: false).last.success?
```

Comment:
not from this PR, but shouldn't we log failure here and below?

Reply:
We don't on this line since its just a check to see if the next line needs to run and the resource not being present isn't an error. But seems reasonable to turn it on for line below.